.github/workflows/release.yml

```diff
@@ -18,126 +10,98 @@ jobs:
       - uses: actions/checkout@v2
       - uses: actions-rs/toolchain@v1
         with:
-          profile: minimal
           toolchain: stable
-          override: true
-
-      # Speed-up runs by using GH cache
-      - uses: actions/cache@v1
-        with:
-          path: ~/.cargo/registry/index
-          key: index-${{ runner.os }}-${{ github.run_number }}
-          restore-keys: |
-            index-${{ runner.os }}-
-      - uses: actions-rs/cargo@v1
-        with:
-          command: generate-lockfile
-      - uses: actions/cache@v1
-        with:
-          path: ~/.cargo/registry/cache
-          key: registry-${{ runner.os }}-${{ steps.rust-version.outputs.version }}-${{ hashFiles('Cargo.lock') }}
-      - uses: actions-rs/cargo@v1
-        with:
-          command: fetch
-      - uses: actions/cache@v1
-        with:
-          path: target
-          key: target-${{ runner.os }}-${{ steps.rust-version.outputs.version }}-${{ hashFiles('Cargo.lock') }}
-
       - uses: actions-rs/cargo@v1
         with:
           command: test
           args: --workspace
-
-  build:
-    name: Build Release
-    needs: test
+  # release:
+  #   name: Create release
+  #   runs-on: ubuntu-latest
+  #   needs:
+  #     - test
```

Comment:
I think we should change it so that `release` requires `test` **and** `build` steps. The `build` steps would require the `test` ones.

Reply:
I think that makes sense. I will refactor the GitHub actions a little, so that:

- Every build step depends on tests. Upon success, upload binary artifact for that run.
- Create release step (that depends on all build steps). This:
  - Creates the release on GitHub
  - Populates every binary by downloading every artifact (every binary for different archs) and uploading it as a release asset.